lib/bolt/transport/winrm/connection.rb

```diff
@@ -183,6 +232,51 @@ def write_executable_from_content(dir, content, filename)
           write_remote_file(content, remote_path)
           remote_path
         end
+
+        private
+
+        # NTSTATUS https://msdn.microsoft.com/en-us/library/cc704588.aspx
+        STATUS_SUCCESS = 0
+        STATUS_LOGON_FAILURE = 0xC000006D
+
+        SMB_PORT = 445
+
+        def smb_client_login
+          return @client if @client
+
+          dispatcher = RubySMB::Dispatcher::Socket.new(smb_socket_connect)
+          @client = RubySMB::Client.new(dispatcher, smb1: true, smb2: true, username: @user, password: target.password)
+          status = @client.login
+          case status.value
+          when STATUS_SUCCESS
+            @logger.debug { "Connected to #{@client.dns_host_name}" }
+          when STATUS_LOGON_FAILURE
+            raise Bolt::Node::ConnectError.new(
+              "SMB authentication failed for #{target.host}",
+              'AUTH_ERROR'
+            )
+          else
+            raise Bolt::Node::ConnectError.new(
+              "Failed to connect to #{target.host} using SMB: #{status.description}",
+              'CONNECT_ERROR'
+            )
+          end
+
+          @client
+        end
+
+        def smb_socket_connect
+          # It's lame that TCPSocket doesn't take a connect timeout
+          # Using Timeout.timeout is bad, but is done elsewhere...
```

Comment:
Yeah, it would be really nice to get rid of `Timeout.timeout` if we can

Reply:
The correct solution for that is detailed in https://stackoverflow.com/a/21014439